charts/rancher-logging/0.0.1/charts/fluentd/templates/service.yaml

```diff
@@ -0,0 +1,30 @@
+{{- if .Values.service }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ template "fluentd.fullname" . }}
+  labels:
+    app: {{ include "fluentd.name" . }}
+    chart: {{ template "fluentd.version" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    kubernetes.io/cluster-service: "true"
+spec:
+  type: {{ .Values.service.type }}
+  ports:
+  {{- range $port := .Values.service.ports }}
+    - name: {{ $port.name }}
+      port: {{ $port.port }}
+      targetPort: {{ $port.port }}
+      {{- if $port.nodePort }}
+      nodePort: {{ $port.nodePort }}
+      {{- end }}
+      {{- if $port.protocol }}
+      protocol: {{ $port.protocol }}
+      {{- end }}
+  {{- end }}
+  selector:
+    app: {{ include "fluentd.name" . }}
+    chart: {{ template "fluentd.version" . }}
```

Comment:
There are so many places using `include`, `include` is the same as `template`, but `include` allow you to bring a template into another template. [ref](https://github.com/technosophos/k8s-helm/blob/master/docs/charts_tips_and_tricks.md#know-your-template-functions)

>  We also added two special template functions: `include` and `required`. The `include` function allows you to bring in another template, and then pass the results to other `template` functions.

Reply:
updated to template